cachi2/core/package_managers/yarn/project.py

```diff
@@ -111,7 +151,8 @@ def is_zero_installs(self) -> bool:
         This is determined by the existence of a non-empty yarn cache folder. For more details on
         zero-installs, see: https://v3.yarnpkg.com/features/zero-installs.
         """
-        return False
+        dir = self.yarn_cache
+        return dir.path.is_dir() and any(file.suffix == ".zip" for file in dir.path.iterdir())
```

Comment:
nitpick: I feel like there's too much going on on this line. Would you mind creating an intermediary variable for the comprehension and then using that inside this boolean expression? IMO it would make it a bit more readable. Having said that I know there's a lot of precedent in the code base so you can happily ignore my comment based on that, but I think despite Python allowing a lot in terms of single-line expressions/constructions, it IMO hinders readability in general.

Reply:
I changed the "is there any zip file" piece to a nested function, that way it's simple to figure out what's happening by reading the function name.

Other alternative is decomposing this to an if block, since the second piece is dependent on the is_dir check. Let me know if you prefer it that way.

Reply:
> Other alternative is decomposing this to an if block, since the second piece is dependent on the is_dir check. Let me know if you prefer it that way.

Call me old-fashioned, but ^this to me sounds like the more readable approach.

Reply:
Ok, changed it to an if block.